### DISCUSSION TITLE
Add "default" attribute value in Attribute class

### DISCUSSION THREAD

**Discussion by antonkomarev** [laravel/framework#50436] [Ideas]
Right now there is ability to specify how to `get` and `set` attribute using the `Illuminate\Database\Eloquent\Casts\Attribute` class:

```php
final class Reaction
{
    protected $attributes = [
        'rate' => self::RATE_DEFAULT,
    ];

    public function rate(): Attribute
    {
        return Attribute::make(
            get: fn (float | null $value) => $value ?? self::RATE_DEFAULT,
            set: function (float | null $value) {
                if ($value !== null && ($value < self::RATE_MIN || $value > self::RATE_MAX)) {
                    throw RateOutOfRange::withValueBetween(
                        $value,
                        self::RATE_MIN,
                        self::RATE_MAX,
                    );
                }

                return $value ?? self::RATE_DEFAULT;
            },
        );
    }
}
```

To specify default value we should define it in the `$attributes` class property array. You could see how it works right now in [Laravel Love package](https://github.com/cybercog/laravel-love).

I propose to add one more optional parameter `default` to the Attribute class:
```php
final class Reaction
{
    public function rate(): Attribute
    {
        return Attribute::make(
            get: fn (float | null $value) => $value ?? self::RATE_DEFAULT,
            set: function (float | null $value) {
                if ($value !== null && ($value < self::RATE_MIN || $value > self::RATE_MAX)) {
                    throw RateOutOfRange::withValueBetween(
                        $value,
                        self::RATE_MIN,
                        self::RATE_MAX,
                    );
                }

                return $value ?? self::RATE_DEFAULT;
            },
            default: fn(float) => self::RATE_DEFAULT,
        );
    }
}
```

Then we will cover all class attribute use cases in one place, make them explicit and allow to dynamically decide what value should be default.

I've tried to add fallback value to `get` as `$value ?? self::RATE_DEFAULT`, but it's not working for all cases.

I found similar proposal, but proposed implementation is different:
- https://github.com/laravel/framework/discussions/43976

One more thing to consider: maybe there is one more improvement to call default from the `get` and `set` methods to not duplicate resolving default value logic.

**Comment by diegogit03** [comment]
This proposal is better, you already started some implementation?

**Reply by antonkomarev** [reply]
Nope, I don't have time for it right now.

**Reply by diegogit03** [reply]
Hi we can try some implementation this weekend? i can help you

**Reply by antonkomarev** [reply]
@diegogit03 sure, just push a PR
